datasets/common_voice/common_voice.py

```diff
@@ -657,55 +659,98 @@ def _info(self):
-        path_to_data = "/".join(["cv-corpus-6.1-2020-12-11", self.config.name])
+        archive_path = dl_manager.download(_DATA_URL.format(self.config.name))
+        if archive_path.startswith("https://"):
+            path_to_data = "/".join(["cv-corpus-6.1-2020-12-11", self.config.name])
+            archive_iterator = dl_manager.iter_archive(archive_path)
+        else:
+            archive_path = dl_manager.extract(archive_path)
+            path_to_data = os.path.join(archive_path, "cv-corpus-6.1-2020-12-11", self.config.name)
+            archive_iterator = None
         path_to_clips = "/".join([path_to_data, "clips"])
 
         return [
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
                 gen_kwargs={
-                    "files": dl_manager.iter_archive(archive),
+                    "archive_iterator": archive_iterator,
                     "filepath": "/".join([path_to_data, "train.tsv"]),
                     "path_to_clips": path_to_clips,
                 },
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.TEST,
                 gen_kwargs={
-                    "files": dl_manager.iter_archive(archive),
+                    "archive_iterator": archive_iterator,
                     "filepath": "/".join([path_to_data, "test.tsv"]),
                     "path_to_clips": path_to_clips,
                 },
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.VALIDATION,
                 gen_kwargs={
-                    "files": dl_manager.iter_archive(archive),
+                    "archive_iterator": archive_iterator,
                     "filepath": "/".join([path_to_data, "dev.tsv"]),
                     "path_to_clips": path_to_clips,
                 },
             ),
             datasets.SplitGenerator(
                 name="other",
                 gen_kwargs={
-                    "files": dl_manager.iter_archive(archive),
+                    "archive_iterator": archive_iterator,
                     "filepath": "/".join([path_to_data, "other.tsv"]),
                     "path_to_clips": path_to_clips,
                 },
             ),
             datasets.SplitGenerator(
                 name="invalidated",
                 gen_kwargs={
-                    "files": dl_manager.iter_archive(archive),
+                    "archive_iterator": archive_iterator,
                     "filepath": "/".join([path_to_data, "invalidated.tsv"]),
                     "path_to_clips": path_to_clips,
                 },
             ),
         ]
 
-    def _generate_examples(self, files, filepath, path_to_clips):
+    def _generate_examples(self, archive_iterator, filepath, path_to_clips):
         """Yields examples."""
+        if archive_iterator is not None:
+            yield from self._generate_examples_streaming(archive_iterator, filepath, path_to_clips)
```

Comment:
The key part: if we get an archive files iterator, then use the new streaming logic, otherwise use the old pre-streaming logic.